kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/kms/aws/KubeAwsKmsTestKmsFacade.java

```diff
@@ -0,0 +1,234 @@
+
+            if (!exists(alias)) {
+                throw new UnknownAliasException(alias);
+            }
+            else {
+                rotate(alias);
+            }
+        }
+
+        @Override
+        public void deleteKek(String alias) {
+            if (!exists(alias)) {
+                throw new UnknownAliasException(alias);
+            }
+            else {
+                delete(alias);
+            }
+        }
+
+        @Override
+        public boolean exists(String alias) {
+            try {
+                read(alias);
+                return true;
+            }
+            catch (KubeClusterException nfe) {
+                return false;
+            }
+        }
+
+        private void create(String alias) {
+            var createKeyResponse = runAwsKmsCommand(CREATE_KEY_RESPONSE_TYPE_REF, awsCmd, KMS, CREATE);
+            kekKeyId = createKeyResponse.keyMetadata().keyId();
+
+            runAwsKmsCommand(awsCmd, KMS, CREATE_ALIAS, PARAM_ALIAS_NAME, ALIAS_PREFIX + alias, PARAM_TARGET_KEY_ID, kekKeyId);
+        }
+
+        private DescribeKeyResponse read(String alias) {
+            return runAwsKmsCommand(DESCRIBE_KEY_RESPONSE_TYPE_REF, awsCmd, KMS, DESCRIBE_KEY, PARAM_KEY_ID, ALIAS_PREFIX + alias);
+        }
+
+        private void rotate(String alias) {
+            // RotateKeyOnDemand is not implemented in localstack.
+            // https://docs.localstack.cloud/references/coverage/coverage_kms/#:~:text=Show%20Tests-,RotateKeyOnDemand,-ScheduleKeyDeletion
+            // https://github.com/localstack/localstack/issues/10723
+            var createKeyResponse = runAwsKmsCommand(CREATE_KEY_RESPONSE_TYPE_REF, awsCmd, KMS, CREATE);
+            kekKeyId = createKeyResponse.keyMetadata().keyId();
+
+            runAwsKmsCommand(awsCmd, KMS, UPDATE_ALIAS, PARAM_ALIAS_NAME, ALIAS_PREFIX + alias, PARAM_TARGET_KEY_ID, kekKeyId);
+        }
+
+        private void delete(String alias) {
+            var key = read(alias);
+            var keyId = key.keyMetadata().keyId();
+            runAwsKmsCommand(SCHEDULE_KEY_DELETION_RESPONSE_TYPE_REF,
+                    awsCmd, KMS, SCHEDULE_KEY_DELETION, PARAM_KEY_ID, keyId, PARAM_PENDING_WINDOW_IN_DAYS, "7" /* Minimum allowed */);
+
+            runAwsKmsCommand(awsCmd, KMS, DELETE_ALIAS, PARAM_ALIAS_NAME, ALIAS_PREFIX + alias);
+        }
+
+        private <T> T runAwsKmsCommand(TypeReference<T> valueTypeRef, String... command) {
+            try {
+                var execResult = runAwsKmsCommand(command);
+                return OBJECT_MAPPER.readValue(execResult.out(), valueTypeRef);
+            }
+            catch (IOException e) {
+                throw new KubeClusterException("Failed to run AWS Kms command: %s".formatted(List.of(command)), e);
+            }
+        }
+
+        private ExecResult runAwsKmsCommand(String... command) {
+            ExecResult execResult;
+            if (awsKmsClient.getClass().equals(AwsKms.class)) {
```

Comment:
this "if" is a code smell.

did you think about two implementations of AbstractAwsKmsTestKmsFacade one for AwsCloud one for AwsLocal.  Push the common stuff to the Abstract class.
Have isAvailable return true on AwsCloud iff the env vars are AWS_CLIENT_ID are set.
Have isAvailable return true on AwsLocal iff the env vars are AWS_CLIENT_ID are not set.